src/agent/src/rpc.rs

```diff
@@ -400,24 +401,46 @@ impl AgentService {
             "exec-id" => eid.clone(),
         );
 
-        let p = sandbox.find_container_process(cid.as_str(), eid.as_str())?;
-
-        let mut signal = Signal::try_from(req.signal as i32).map_err(|e| {
+        let sig = Signal::try_from(req.signal as i32).map_err(|e| {
             anyhow!(e).context(format!(
                 "failed to convert {:?} to signal (container-id: {}, exec-id: {})",
                 req.signal, cid, eid
             ))
         })?;
 
+        if eid.is_empty() {
+            // eid is empty, signal all the processes in the container cgroup
+            let ctr = sandbox
+                .get_container(&cid)
+                .ok_or_else(|| anyhow!("Invalid container id"))?;
+            let cm = if ctr.cgroup_manager.is_some() {
+                ctr.cgroup_manager.as_ref().unwrap()
+            } else {
+                return Err(anyhow!("container cgroup not exist"));
+            };
+
+            let pids = cm.get_pids()?;
```

Comment:
Do we need to freeze the cgroup manager, like runc shimv2 dose? Otherwise it looks like there can be a race window that some process forked another process after `cm.get_pids()` and then we'll fail to kill it and leak it.

Another option might be that we send `kill(-pid, signal)` to all processes returned by the cgroup manager.

Reply:
@bergwolf Thanks. I have updated the change following your suggestion.